app/controllers/api/v1/spine_terms_controller.rb

```diff
@@ -14,9 +14,23 @@ def index
         includes = %i(property)
         includes += %i(vocabularies) unless params[:spine].present?
         includes += %i(organization) if params[:with_organization].present?
-        terms = Spine.find(params[:id]).terms.includes(includes)
 
-        render json: terms, spine: params[:with_weights].present?,
+        max_weight = current_configuration_profile.predicates.maximum(:weight) || 0
+
+        select = "terms.*, " \
```

Comment:
do we need this all the time or only with_weight param? 
also looks that we can calculate and save at model current mapping weight, same as for max weight. it'll not be changed often.